File: 01_basic/week2/344_reverse-string_watermelon.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        
        # 좌측 끝과 우측 끝을 차례대로 swap
        for index in range(len(s) // 2):
            s[index], s[len(s) - index - 1] = s[len(s) - index - 1], s[index]

File: 01_basic/week2/344_reverse-string_watermelon.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    for (let i = 0; i < Math.floor(s.length / 2); i++) {
        [s[i], s[s.length - i - 1]] = [s[s.length - i - 1], s[i]];  // Destructuring assignment
    }
    
    // s.reverse()도 가능
};
